cmux-tui/bindings/go/raw/client.go

```
@@ -222,18 +221,18 @@ func ResolveSocketPath(explicit, session string) (string, error) {
 
 // ValidateSession rejects names that could escape the private runtime
-// directory or cannot be used portably by the server.
+// directory or carry control text into the socket path.
 func ValidateSession(session string) error {
-	if !validSessionName.MatchString(session) || session == "." || session == ".." {
-		return fmt.Errorf(
-			"%w: session must match [A-Za-z0-9][A-Za-z0-9._-]{0,63}",
-			ErrInvalidArgument,
-		)
+	if err := sessionpath.Validate(session); err != nil {
+		return fmt.Errorf("%w: %v", ErrInvalidArgument, err)
```

golangci-lint (2.12.2)

[error] non-wrapping format verb for fmt.Errorf. Use `%w` to format errors

(errorlint)